api.go

```diff
@@ -119,6 +123,7 @@ type AuthorizationConfig struct {
 	Config                     // the base config
 	HelpMessage string         // additional message for the user on internal authz errors
 	Token       IdentityToken  // the token provider for calls to Athenz
+	AthenzX509  AthenzX509     // the x509 provider for calls to Athenz
```

Comment:
if we add a `authorizationMode=TOKEN|TLS` and call the appropriate client() method, would it be more straightforward?

Reply:
I was considering to add `X509Mode` as a boolean property so that it is easy to match instead of string-based flag. what do you think?

Reply:
X509Mode might be confusing when we add a feature flag to support user token/x509 certs. Let's add a prefix like `Identity` to this flag to be explicit about where it's used